PyInstaller/building/utils.py

```diff
@@ -492,6 +492,20 @@ def format_binaries_and_datas(binaries_or_datas, workingdir=None):
             src_root_paths = glob.glob(src_root_path_or_glob)
 
         if not src_root_paths:
+            
+            # on Debian/Ubuntu, a missing pyconfig.h can be fixes with
+            # installing python-dev
+            if 'pyconfig.h' in src_root_path_or_glob:
+                msg = """Unable to find "%s" when adding binary and data files.
+This would mean your Python installation doesn't come with proper library files.
+This usually happens by missing development package, or unsuitable build parameters of Python installation.
+* On Debian/Ubuntu, you would need to install Python development packages
+  * apt-get install python3-dev
+  * apt-get install python-dev
+* If you're building Python by yourself, please rebuild your Python with `--enable-shared` (or, `--enable-framework` on Darwin)
+""" % src_root_path_or_glob
```

Comment:
Having two `raise` statements side-by side is a bit hard to read IMHO. Also the text of the message is duplicate. How about this:
```
if not src_root_paths:
    msg = 'Unable to find "%s" when adding binary and data files.' % (
            src_root_path_or_glob))
    if src_root_path_or_glob.endswith(…)
        msg += """\n…"""
    raise SystemExit(msg)
```
 What do you think?

Reply:
I agree.  Should I do the `raise` with an empty line before it?  I think that might be more readable:

```python
if not src_root_paths:
    msg = 'Unable to find "%s" when adding binary and data files.' % (
                  src_root_path_or_glob))
    if src_root_path_or_glob.endswith("pyconfig.h"):
        msg += """blah
...
blah"""
        
    raise SystemExit(msg)

Reply:
If any I'd prefer an empty line behind. But both may fail the linter :-}

Reply:
I'll pray to the lint gods that it passes!